backend/apollo/elections/serializers.py

```diff
@@ -85,10 +86,16 @@ def validate(self, attrs):
         try:
             rule_type = getattr(VoterAuthorizationRule.Type, rule_type)
         except AttributeError:
-            raise serializers.ValidationError("INVALID_RULE_TYPE")
-        else:
-            attrs["type"] = rule_type
-            return super().validate(attrs)
+            raise serializers.ValidationError({"type": "INVALID_RULE_TYPE"})
+
+        attrs["type"] = rule_type
+        if attrs["type"] == VoterAuthorizationRule.Type.REGEX:
+            try:
+                re.compile(attrs["value"])
+            except Exception:
+                raise serializers.ValidationError({"value": "INVALID_REGEX"})
```

Comment:
Ultimately it could be a good idea to delegate the validation to some class, but for now whatever.